src/TimingData.h

```diff
@@ -27,36 +27,182 @@ inline T ToDerived( const TimingSegment *t, TimingSegmentType tst )
 	return static_cast<T>( t );
 }
 
-#define TimingSegmentToXWithName(Seg, SegName, SegType) \
-	inline const Seg* To##SegName( const TimingSegment *t ) \
```

Comment:
I think you can create inline templated functions here, which I think is nicer.

Reply:
> Yeah, I'd also like that a lot better. Based on this and your comment on the other macro changes PR, and the age of this PR (it would cause conflicts in notefield if i update it now) I will make a new one that's better organized and has templates instead of expanded inlines.

_Originally posted by @sukibaby in https://github.com/itgmania/itgmania/pull/400#discussion_r1769467262_